startupscript/butane/idle-shutdown.sh

```diff
@@ -18,12 +19,16 @@ readonly -f emit
 # shellcheck source=/dev/null
 source /home/core/metadata-utils.sh
 
-# Get the idle timeout in seconds. By default, the VM timeout after 2 days of continued idleness.
-IDLE_TIMEOUT_SECONDS="${1:-172800}"
+IDLE_TIMEOUT_SECONDS="$(get_metadata_value "idle-timeout-seconds")"
+readonly IDLE_TIMEOUT_SECONDS
+if [[ -z "${IDLE_TIMEOUT_SECONDS}" ]]; then
+    emit "No idle timeout seconds set. Do not autostop VM."
+    exit 0
+fi
 
 # Get the last time the VM was active.
-LAST_ACTIVE="$(get_guest_attribute "last-active/cpu" || echo "0")"
-LAST_ACTIVE_PROXY="$(get_guest_attribute "last-active/proxy" || echo "0")"
+LAST_ACTIVE="$(get_guest_attribute "last-active/cpu")"
```

Comment:
this used to be 0 when not set, and now its "".  seems like maybe that will break the compare below?

Reply:
actually <some value> -lt "" return false and doesn't throw error

Reply:
It would be better not to depend on unintuitive or infrequently exercised behavior like: `value1 -lt ""`.

It would be better to have the caller explicitly reinterpret the empty string to 0.
So either having the caller pass a default to get_metadata_value() or doing something like:

```
LAST_ACTIVE="${LAST_ACTIVE:-0}"
LAST_ACTIVE_PROXY="${LAST_ACTIVE_PROXY:-0}"
```

to set the values to 0 if they are empty.

Reply:
added default value.